File: ruby/count_occurrences.rb
test_cases = ["the three truths", "th"], ["ababababab", "abab"]

def count_occurrences(string, substring)
  string.split(substring).length - 1
end

test_cases.each { |test_case| p count_occurrences(test_case[0], test_case[1]) }

File: nim/countoccurrences.nim
import strutils

let testCases = @[
  @["the three truths", "th"],
  @["ababababab", "abab"]
]

proc countOccurrences(target: string, substring: string): int =
  target.split(substring).len - 1

for testCase in testCases:
  echo countOccurrences(testCase[0], testCase[1])
